core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceCreator.scala

```diff
@@ -0,0 +1,688 @@
+              dataAccess.methodConfigurationQuery.create(destWorkspace, methodConfig.get)
+            }
+          }
+          DBIO.seq(inserts: _*)
+        } andThen {
+        DBIO.successful()
+      }
+    }
+  }
+
+  /**
+    * Find all the objects in sourceWorkspace's bucket that start with filePrefix and copy them to destWorkspace's
+    * bucket
+    *
+    * @param filePrefix
+    * @param sourceWorkspace
+    * @param destWorkspace
+    */
+  private def copyBucketContents(sourceWorkspace: Workspace, destWorkspace: Workspace, filePrefix: Option[String]): Future[List[Option[StorageObject]]] = {
+    filePrefix match {
+      case Some(prefix) => gcsDAO.listObjectsWithPrefix(sourceWorkspace.bucketName, prefix).flatMap { objectsToCopy =>
+        Future.traverse(objectsToCopy) { objectToCopy => gcsDAO.copyFile(sourceWorkspace.bucketName, objectToCopy.getName, destWorkspace.bucketName, objectToCopy.getName) }
+      }
+      case None => Future.successful(List[Option[StorageObject]]())
+    }
+  }
+
+  /**
+    * Use this method to make all the checks with Sam about whether the user has the required permissions to create a
+    * Workspace in this Namespace/Billing Project.  These checks should throw RawlsExceptionsWithErrorReport any time
+    * they encounter a scenario where the user does not have the necessary permissions.
+    *
+    * @param workspaceRequest
+    * @param parentSpan
+    * @return
+    */
+  private def checkCreateWorkspacePermissions(workspaceRequest: WorkspaceRequest, parentSpan: Span): Future[Unit] = {
+    // Running futures in parallel
+    val requireCreateWorkspaceAccessFuture = requireCreateWorkspaceAccess(workspaceRequest, parentSpan)
+    val maybeRequireBillingProjectOwnerAccessFuture = maybeRequireBillingProjectOwnerAccess(workspaceRequest, parentSpan)
+
+    for {
+      _ <- requireCreateWorkspaceAccessFuture
+      _ <- maybeRequireBillingProjectOwnerAccessFuture
+    } yield ()
+  }
+
+  /**
+    * Make an API call to Sam to check if the user has permissions to create a Workspace in the specified Billing
+    * Project.  Throw an RawlsExceptionWithErrorReport if user does not have required permissions.
+    *
+    * @param workspaceRequest
+    * @param parentSpan
+    * @return
+    */
+  private def requireCreateWorkspaceAccess(workspaceRequest: WorkspaceRequest, parentSpan: Span = null): Future[Boolean] = {
+    val projectName = RawlsBillingProjectName(workspaceRequest.namespace)
+    traceWithParent("checkUserCanCreateWorkspace", parentSpan)(_ => samDAO.userHasAction(SamResourceTypeNames.billingProject, projectName.value, SamBillingProjectActions.createWorkspace, userInfo)).map {
+      case true => true
+      case false => throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, s"You are not authorized to create a workspace in billing project ${workspaceRequest.toWorkspaceName.namespace}"))
+    }
+  }
```

Comment:
For consistency, these util methods should have the same signature - other methods return `Future[Unit]`, this one returns a `Boolean` but can only ever return true